addons/mail/models/ir_qweb.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+from odoo import models
+
+class QWebCodeFound(Exception):
+    """
+    Exception raised when a qweb compilation encounter dynamic content if the
+    option `raise_on_code` is True.
+    """
+
+class IrQweb(models.AbstractModel):
+    _inherit = 'ir.qweb'
+
+    def _get_template_cache_keys(self):
+        return super()._get_template_cache_keys() + ['raise_on_code']
+
+    def _compile_directives(self, el, options, indent):
+        if options.get('raise_on_code'):
+            raise QWebCodeFound()
+        return super()._compile_directives(el, options, indent)
```

Comment:
I left the feature, but for me it has absolutely nothing to do in qweb. It's just a parsing of the template to see if there are any `t-` attributes. The QWeb is made to execute directives otherwise it should not be used.